domain/ide/workspace/folder.go

```diff
@@ -193,17 +193,20 @@ func (f *Folder) processResults(scanData snyk.ScanData) {
 	// TODO: perform issue diffing (current <-> newly reported)
 	// Update diagnostic cache
 	for _, issue := range scanData.Issues {
-		cachedIssues, _ := f.documentDiagnosticCache.Load(issue.AffectedFilePath)
+		uniqueIssueId := f.getUniqueIssueID(issue)
+
+		cachedIssues, _ := f.documentDiagnosticCache.Load(uniqueIssueId)
```

Comment:
I'm not sure this is going to work, but let's chat so I can understand the whole picture. The access path to cached diagnostics is usually the file name, so I wonder how that works with this change. Usually, when you ask for the diagnostics, you know for which file you want them, but not what they are.

Reply:
Oh, I see my mistake now. My initial assumptions led me down the wrong path 😞

I've been reading https://github.com/puzpuzpuz/xsync and maybe a better approach would be to keep the file path as the key for the diagnostics cache, and handle the deduplication logic separately?

Happy to chat!!

Thank you 🙌